src/parser/core/modules/ActionWindow/windows/RaidBuffWindow.ts

```diff
@@ -0,0 +1,79 @@
+import {Event, Events} from 'event'
+import {BuffWindow, EvaluatedAction, PlayersBuffedEvaluator} from 'parser/core/modules/ActionWindow'
+import {HistoryEntry} from 'parser/core/modules/ActionWindow/History'
+import {ensureArray} from 'utilities'
+import {filter, oneOf, noneOf} from '../../../filter'
+
+const FULL_PARTY = 8
+
+/**
+ * Tracks buffs applied to the party.
+ */
+export abstract class RaidBuffWindow extends BuffWindow {
+
+	/**
+	 * Array of all status apply events on players for the buffs specified.
+	 *
+	 * These might ideally belong in the History, but it would break the
+	 * current paradigm of the ActionWindow.
+	 */
+	protected raidBuffApplications = new Array<Events['statusApply']>()
+
+	/**
+	 * Expected number of people who should get the raid buff every time.
+	 */
+	protected expectedCount = FULL_PARTY
```

Comment:
You get the full party in `partyMembers`, could you not use `length` on that to get the actual party size?